force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,121 @@
+        Lead testLead2 = new Lead(FirstName = 'John', LastName = 'Doe', Company = 'Cloud Code', Title = 'head of department');
+        Lead testLead3 = new Lead(FirstName = 'Jill', LastName = 'Doe', Company = 'Cloud Code', Title = 'chief');
+        Lead testLead4 = new Lead(FirstName = 'Jim', LastName = 'Doe', Company = 'Cloud Code', Title = 'assistant finance officer');
+
+        testLeads.add(testLead1);
+        testLeads.add(testLead2);
+        testLeads.add(testLead3);
+        testLeads.add(testLead4);
+
+        Test.startTest();
+        insert testLeads;
+        Test.stopTest();
+        
+        List <Lead> insertedTestLeads = [SELECT Id, Name, Title FROM Lead WHERE Id IN :testLeads];
+        Set <String> leadTitles = new Set <String> ();
+        for (Lead ld : insertedTestLeads){
+            leadTitles.add(ld.Title);
+        }
+
+        Assert.areEqual(true, leadTitles.contains('Vice President'), 'Title should be Vice President');
+        Assert.areEqual(true, leadTitles.contains('Manager'), 'Title should be Manager');
+        Assert.areEqual(true, leadTitles.contains('Executive'), 'Title should be Executive');
+        Assert.areEqual(true, leadTitles.contains('Assistant'), 'Title should be Assistant');
+    }
+
+    @isTest
+    public static void LeadTriggerHandler_handleAutoLeadScoring(){
+        List <Lead> testLeadsWithScores = new List <Lead> ();
+        Lead testLead5 = new Lead(FirstName = 'Jane', LastName = 'Doe', Company = 'Cloud Code', LeadSource = 'Website', Phone = '6478355589', Email = 'jane.doe@cloudcode.com', Industry = 'Technology', Title = 'vp');
+        Lead testLead6 = new Lead(FirstName = 'Jill', LastName = 'Doe', Company = 'Cloud Code', LeadSource = 'Website', Phone = '6478355589', Email = 'jill.doe@cloudcode.com', Title = 'vp');
+        Lead testLead7 = new Lead(FirstName = 'John', LastName = 'Doe', Company = 'Cloud Code', LeadSource = 'Website', Email = 'john.doe@cloudcode.com', Title = 'vp');
+        Lead testLead8 = new Lead(FirstName = 'Jim', LastName = 'Doe', Company = 'Cloud Code', Phone = '6478355589', Title = 'vp');
+
+        testLeadsWithScores.add(testLead5);
+        testLeadsWithScores.add(testLead6);
+        testLeadsWithScores.add(testLead7);
+        testLeadsWithScores.add(testLead8);
+
+        Test.startTest();
+        insert testLeadsWithScores;
+        Test.stopTest();
+
+        List <Lead> insertedTestLeads = [SELECT Id, Lead_Score__c FROM Lead WHERE Id IN :testLeadsWithScores];
+        Set <Decimal> leadScores = new Set <Decimal> ();
+        for (Lead ld : insertedTestLeads){
+            leadScores.add(ld.Lead_Score__c);
+        }
+
+        Assert.areEqual(true, leadScores.contains(18), 'Score should be 18');
+        Assert.areEqual(true, leadScores.contains(8), 'Score should be 8');
+        Assert.areEqual(true, leadScores.contains(3), 'Score should be 3');
+        Assert.areEqual(true, leadScores.contains(5), 'Score should be 5');
+    }
```

Comment:
This is great that you're checking that these values are in there but you should also check that these values are where they are supposed to be. Maybe check with the first names to see if Jill is the right one. Also, there is no null checks. Always try to do positive and negative test cases.